includes/compat/subscriptions/class-wc-payment-gateway-wcpay-subscriptions-compat.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Class WC_Payment_Gateway_WCPay_Subscriptions_Compat
+ *
+ * @package WooCommerce\Payments
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+use WCPay\Logger;
+
+/**
+ * Gateway class for WooCommerce Payments, with added compatibility with WooCommerce Subscriptions.
+ */
+class WC_Payment_Gateway_WCPay_Subscriptions_Compat extends WC_Payment_Gateway_WCPay {
+
+	/**
+	 * WC_Payment_Gateway_WCPay_Subscriptions_Compat constructor.
+	 *
+	 * @param array ...$args Arguments passed to the main gateway's constructor.
+	 */
+	public function __construct( ...$args ) {
+		parent::__construct( ...$args );
+
+		$this->supports = array_merge(
+			$this->supports,
+			[
+				'subscriptions',
+				'subscription_cancellation',
+				'subscription_suspension',
+				'subscription_reactivation',
+				'subscription_amount_changes',
+				'subscription_date_changes',
+				'subscription_payment_method_change',
+				'subscription_payment_method_change_customer',
+				'subscription_payment_method_change_admin',
+				'multiple_subscriptions',
```

Comment:
Some of those capabilities are a lie at this moment. If we are going to do a release with only partial subscriptions support (for example, not allowing changing the payment method), let's remember to remove the appropiate keys.

Reply:
@RadoslavGeorgiev - would you add to the PR description a checklist to test each of these (and remove any that don't pass)

Reply:
Done, see PR description 🎉

Reply:
OK, this PR is big enough. I think we should
- ship this PR with just the first three already verified
- add subscriber initiated payment method change support using @luizreis PR #832 
- and defer the others so we can ship 1.4.0

Agreed? cc @thenbrent @RadoslavGeorgiev @reykjalin @luizreis

Reply:
Sounds good!

I'd just remove this capability before shipping 1.4:

* `subscription_payment_method_change_admin`

The rest of the capabilities should be fine to keep.

These two will be handled by #832:

* `subscription_payment_method_change` (I don't actually think this is required as it's an alias for `subscription_payment_method_change_customer`)
* `subscription_payment_method_change_customer`

These other capabilities shouldn't require any extra work with WC Pay as it uses tokens to recurring payments and these caps only require special handling for gateways like PayPal Standard, where there is a subscription plan managed on the gateway's side that needs to be kept in-sync with WooCommerce:

* `subscription_reactivation`
* `subscription_amount_changes`
* `subscription_date_changes`
* `multiple_subscriptions`